src/Aspire.Hosting/Health/ResourceHealthCheckService.cs

```diff
@@ -342,26 +342,16 @@ internal async Task<bool> DelayAsync(ResourceEvent? currentEvent, TimeSpan delay
                     _logger.LogTrace("Health monitoring delay interrupted for resource '{Resource}'.", _resourceName);
                     return true;
                 }
-                if (_delayCts == null || !_delayCts.TryReset())
-                {
-                    _delayCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
-                }
                 _delayInterruptTcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+                delayInterruptedTask = _delayInterruptTcs.Task;
             }
 
-            var completedTask = await Task.WhenAny(Task.Delay(delay, _delayCts.Token), _delayInterruptTcs.Task).ConfigureAwait(false);
+            // Don't throw to avoid writing the thrown exception to the debug console.
+            // See https://github.com/dotnet/aspire/issues/7486
```

Comment:
can't they get this from the change description? then the codebase doesn't clutter up with old close issue URLs

... but if this is team style, fair enough

Reply:
We do it occassionally in aspnetcore and here.

This isn't a problem that's usually considered (exceptions notification in debug console) which is why I linked to the issue.